core/src/main/java/org/apache/iceberg/view/BaseViewCatalog.java

```diff
@@ -0,0 +1,190 @@
+  public View loadView(TableIdentifier identifier) {
+    if (isValidIdentifier(identifier)) {
+      ViewOperations ops = newViewOps(identifier);
+      if (ops.current() == null) {
+        throw new NoSuchViewException("View does not exist: %s", identifier);
+      } else {
+        return new BaseView(newViewOps(identifier), ViewUtil.fullViewName(name(), identifier));
+      }
+    }
+
+    throw new NoSuchViewException("Invalid view identifier: %s", identifier);
+  }
+
+  @Override
+  public ViewBuilder buildView(TableIdentifier identifier) {
+    return new BaseViewBuilder(identifier);
+  }
+
+  protected class BaseViewBuilder implements ViewBuilder {
+    private final TableIdentifier identifier;
+    private final ImmutableViewVersion.Builder viewVersionBuilder = ImmutableViewVersion.builder();
+    private final List<ViewRepresentation> viewRepresentations = Lists.newArrayList();
+    private final Map<String, String> properties = Maps.newHashMap();
+    private Schema schema;
+
+    protected BaseViewBuilder(TableIdentifier identifier) {
+      Preconditions.checkArgument(
+          isValidIdentifier(identifier), "Invalid view identifier: %s", identifier);
+      this.identifier = identifier;
+    }
+
+    @Override
+    public ViewBuilder withSchema(Schema newSchema) {
+      this.schema = newSchema;
+      viewVersionBuilder.schemaId(newSchema.schemaId());
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withQuery(String dialect, String sql) {
+      viewRepresentations.add(
+          ImmutableSQLViewRepresentation.builder().dialect(dialect).sql(sql).build());
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withDefaultCatalog(String defaultCatalog) {
+      viewVersionBuilder.defaultCatalog(defaultCatalog);
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withDefaultNamespace(Namespace namespace) {
+      viewVersionBuilder.defaultNamespace(namespace);
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withProperties(Map<String, String> newProperties) {
+      this.properties.putAll(newProperties);
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withProperty(String key, String value) {
+      this.properties.put(key, value);
+      return this;
+    }
+
+    @Override
+    public View create() {
+      ViewOperations ops = newViewOps(identifier);
+      if (null != ops.current()) {
+        throw new AlreadyExistsException("View already exists: %s", identifier);
+      }
+
+      ViewVersion viewVersion =
+          viewVersionBuilder
+              .versionId(1)
+              .addAllRepresentations(viewRepresentations)
+              .timestampMillis(System.currentTimeMillis())
+              .putSummary("operation", "create")
+              .build();
+
+      ViewMetadata viewMetadata =
+          ViewMetadata.builder()
+              .setProperties(properties)
+              .setLocation(defaultWarehouseLocation(identifier))
```

Comment:
You can't set the location?

Reply:
Not sure I'm understanding exactly what you mean here. For `TableMetadata` we're setting the location similarly in https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java#L193 or do you mean the fact that there's no method on the `ViewBuilder` that would allow setting a location?